front/app/components/SignUpIn/SignUp/PasswordSignup.tsx

```diff
@@ -297,30 +297,39 @@ class PasswordSignup extends PureComponent<Props & InjectedIntlProps, State> {
     }
   };
 
-  handleOnSubmit = async (event: React.FormEvent) => {
```

Comment:
Gave handleOnSubmit a bit less responsibility and extracted some functionality into other functions.